### PR TITLE
sparqljs: accurate module representation

### PR DESCRIPTION
**Pull request by tpluscode**
`import * as Sparqljs from 'sparqljs'` is wrong because the package in CJS and default-import like `import Sparqljs from 'sparqljs'` must be used

In this PR I adjusted the index to reflect that> I only moved `Wildcard` to its own respective module because it is independent of the other types

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Review by rubensworks — CHANGES_REQUESTED** [review]
There appear to be some breaking changes in this PR.
Could we keep thing backwards-compatible?

I get the following errors when compiling comunica:

```
node_modules/@types/sparqljs/index.d.ts:9:27 - error TS2307: Cannot find module './lib/Wildcard.js' or its corresponding type declarations.

9 import Wildcard = require('./lib/Wildcard.js');
                            ~~~~~~~~~~~~~~~~~~~

node_modules/sparqlalgebrajs/lib/algebra.d.ts:113:15 - error TS2749: 'Wildcard' refers to a value, but is being used as a type here. Did you mean 'typeof Wildcard'?

113     wildcard: Wildcard;
                  ~~~~~~~~

node_modules/sparqlalgebrajs/lib/util.d.ts:86:43 - error TS2749: 'Wildcard' refers to a value, but is being used as a type here. Did you mean 'typeof Wildcard'?

86     static isWildcard(term: any): term is Wildcard;
```

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/sparqljs/index.d.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against cd91ea2da78138a106b60861cebc91a8300e587b
</p>

File: types/sparqljs/index.d.ts [hunk 1]
```diff
@@ -1,303 +1,311 @@
...
-}
-
-export interface AggregateExpression extends BaseExpression {
-    type: "aggregate";
-    expression: Expression;
-    aggregation: string;
-    separator?: string | undefined;
-}
+// tslint:disable-next-line:no-unnecessary-class
+declare class SparqlJs {
```
Comment (jakebailey) [inline types/sparqljs/index.d.ts:4]: This doesn't look right; upstream has `module.exports = { ... }` with just three things, not a class exported with extra props: https://unpkg.com/browse/sparqljs@3.7.1/sparql.js

If this is also related to `import/default`, I'm pretty sure that lint rule is bugged if it's complatining. https://arethetypeswrong.github.io/?p=sparqljs%403.7.1 is happy.

**Review by jakebailey — CHANGES_REQUESTED** [review]

File: types/sparqljs/index.d.ts (hunk 1 above)
Comment (jakebailey) [reply types/sparqljs/index.d.ts:4]: I meant to comment on `export =`, oops. Still, point stands.

**Review by rubensworks — APPROVED** [review]
Seems to work now on my end! :-)

**Comment by rubensworks** [comment]
@jakebailey It broke for me here: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/67040#pullrequestreview-1684283993

**Comment by jakebailey** [comment]
So that was just about the temporary state of the PR and not the current published package?

**Pull request closed without merging**